### ISSUE TITLE
getMedia() returning null, even though hasMedia() returns true on hashtag->getFeed()-getItems() objects

### ISSUE THREAD

**Issue by donotnoot** [mgp25/Instagram-API#1886]
### Before submitting an issue make sure you have:
- [x] [Searched](https://github.com/mgp25/Instagram-API/search?type=Issues) the bugtracker for similar issues including **closed** ones
- [x] [Read the FAQ](https://github.com/mgp25/Instagram-API/wiki/FAQ)
- [x] [Read the wiki](https://github.com/mgp25/Instagram-API/wiki)
- [x] [Reviewed the examples](https://github.com/mgp25/Instagram-API/tree/master/examples)
- [x] [Installed the api using ``composer``](https://github.com/mgp25/Instagram-API#installation)

### Purpose of your issue?
- [x] Bug report (encountered problems/errors)
- [ ] Feature request (request for new functionality)
- [x] Question
- [ ] Other

---

```php
	// Start IG Session
	try{
		$ig->login(
			conf('ig.username'),
			conf('ig.password')
		);
	} catch (Exception $e) {

	}

	// Browse hashtag page
	$uuid = \InstagramAPI\Signatures::generateUUID();
	$ht = $ig->hashtag->getFeed("dog", $uuid); // Any hashtag is doing this though
	$results = $ht->getItems();
	$results[0]->printJson();
	var_dump($results[0]->hasMedia());
	var_dump($results[0]->getMedia());
```
The above code outputs

```
{
    "taken_at": "1522540608",
    "pk": "1747519742740163608",
    "id": "1747519742740163608_2221482432",
    "device_timestamp": "1522540605139966",
    "media_type": 1,
    "code": "BhAb_KAAmwY",
    "client_cache_key": "MTc0NzUxOTc0Mjc0MDE2MzYwOA==.2",
    "filter_type": 0,
    "image_versions2": {
        "candidates": [
       [........]
    "has_more_comments": false,
    "max_num_visible_preview_comments": 2,
    "preview_comments": [],
    "comment_count": 0,
    "photo_of_you": false,
    "can_viewer_save": true,
    "organic_tracking_token": "eyJ2ZXJzaW9uIjo1LCJwYXlsb2FkIjp7ImlzX2FuYWx5dGljc190cmFja2VkIjp0cnVlLCJ1dWlkIjoiZmM5YTEyNGYyMDI2NDNlZWI1YjBhZjdmNDc0Y2M4N2UxNzQ3NTE5NzQyNzQwMTYzNjA4Iiwic2VydmVyX3Rva2VuIjoiMTUyMjU0MDYxMjU2N3wxNzQ3NTE5NzQyNzQwMTYzNjA4fDczOTc2MTkyNTh8ZWY5YjQ2ODdhNTI3YzE1MjUxZTYyODFlZGUzM2I1MDE3Nzk5M2UxZmZhNzkzZGY3NTZlM2NjMGEyOWIxMmMzOSJ9LCJzaWduYXR1cmUiOiIifQ=="
}
bool(true)
NULL
```
---

So basically, I'm either missing something obvious (possibly sleep too) or there is some sort of bug in the API. 
When I get a hashtag feed and try to get the media from any of its items getMedia() returns NULL even thoguh hasMedia(); returns true. I tried with several HT's and it seems to be doing the same with all.
If I call printJson() on $results[any], a nice JSON object shows up with all the info from the post.

I hope that I'm not missing something obvious but I read the FAQ and the Wiki and can't seem to find why it's doing this.

Thanks

**Comment by donotnoot** [comment]
So it was obviously my fault, the result array are of class Item, and you don't need to call getMedia() on them.

I was confused because on the accessingValues.php example it calls getMedia on the results from the explore feed before getting the post info and all that.

Oh well, at least I figured it out on my own.

**Comment by aitte2** [comment]
Hi. How did you find the `has*` function? It is not documented in this API, probably because it doesn't do what you think it does. ;) It checks if a property named "media" is valid for that object. It doesn't say if it has any value. At least I think so from this message linked from the release notes: https://github.com/mgp25/Instagram-API/issues/1626#issuecomment-337741906

**Comment by donotnoot** [comment]
Hi, 

Now that you mention that, it makes sense. I think it popped up when I called  `printPropertyDescriptions()` on the result Item from `$ig->hashtag->getFeed($hashtag, $uuid)->getItems()`.

One thing I don't fully understand is why you have to call `getMedia()` on Item objects from `$ig->discover->getExploreFeed()->getItems()` but not on the HT feed function results (above).

Since the method is called `getItems()` on both I assumed you have to treat the Item objects that they return in the same way. (?)

I might be missing something, I don't know.

Thanks for replying

**Comment by aitte2** [comment]
Hi. Okay that makes sense. I guess it's listed in `printPropertyDescriptions()`. Well you know what it does now :)

And about getItems:

- https://github.com/mgp25/Instagram-API/blob/master/src/Response/TagFeedResponse.php has `'items'                  => 'Model\Item[]',`
- https://github.com/mgp25/Instagram-API/blob/master/src/Response/ExploreResponse.php has `'items'                  => 'Model\ExploreItem[]',`
- And an exploreitem is: https://github.com/mgp25/Instagram-API/blob/master/src/Response/Model/ExploreItem.php which has `'media'             => 'Item',`

**Comment by donotnoot** [comment]
Well, that is extremely clear now.  I've been very familiar with PHP for 5 or 6 years but always avoided OOP and coded everything procedurally (in all languages) for some reason, so it often confuses me. Probably about time to work on that...

Thank you very much for taking the time to reply!

**Comment by aitte2** [comment]
Hehe no problem :) 👍
